### PR TITLE
chore: Apply minor code quality revisions

### PR DESCRIPTION
**Pull request by akshgpt7**
# Description
I ran [DeepSource](https://deepsource.io/) analysis on my fork of the repo, and found some interesting [code quality issues](https://deepsource.io/gh/akshgpt7/pyhf/) that can improve the general code quality here.
This PR fixes a few of the issues detected.

### Summary of changes:
- Refactor unnecessary use of `elif/else` after `return`.
- Simplify `if` statements to use `bool()`.
- Refactor iterating over dictionary instead of using `dict.keys()`.
- Refactor unnecessary `dict()` call.

# Checklist Before Requesting Reviewer

- [x] Tests are passing
- [x] "WIP" removed from the title of the pull request
- [x] Selected an Assignee for the PR to be responsible for the log summary

# Before Merging

For the PR Assignees:

- [x] Summarize commit messages into a comprehensive review of the PR

```
* Apply minor code changes
   - Use raise error as default behavior to avoid if-else evaluation
   - Use evaluation of bool() over trivial if statements
   - Drop use of keys on dict iteration
```

### CODE REVIEW

**Comment by akshgpt7** [comment]
@matthewfeickert The test seems to be failing for macOS when it tries to install tensorflow. Can you help me know if this is a fixable error?

**Comment by matthewfeickert** [comment]
Thanks for your PR @akshgpt7! We'll take a look at this later today and get it reviewed, which should be easy enough. We're probably not interested in picking up additional config files or dependencies at the moment though, so right off the bat can we ask you to remove `.deepsource.toml`?

> The test seems to be failing for macOS ...

The VMs for GHA sometimes can be finicky for Mac. I restarted the CI.

**Comment by akshgpt7** [comment]
@matthewfeickert removed the `.deepsource.toml` file as requested.

File: src/pyhf/optimize/__init__.py [hunk 1]
```diff
@@ -15,7 +15,7 @@ def __getattr__(self, name):
             # for autocomplete and dir() calls
             self.scipy_optimizer = scipy_optimizer
             return scipy_optimizer
-        elif name == 'minuit_optimizer':
+        if name == 'minuit_optimizer':
```
Comment (matthewfeickert) [inline src/pyhf/optimize/__init__.py:18]: This doesn't really do anything to justify the change.

File: src/pyhf/tensor/__init__.py [hunk 2]
```diff
@@ -10,7 +10,7 @@ def __getattr__(self, name):
             # for autocomplete and dir() calls
             self.numpy_backend = numpy_backend
             return numpy_backend
-        elif name == 'jax_backend':
+        if name == 'jax_backend':
```
Comment (matthewfeickert) [inline src/pyhf/tensor/__init__.py:13]: This doesn't really do anything to justify the change.

File: tests/test_export.py [hunk 3]
```diff
@@ -274,7 +274,7 @@ def test_export_bad_modifier(caplog):
     "spec, normfactor_config",
     [
         (spec_staterror(), dict(name='mu', inits=[1.0], bounds=[[0.0, 8.0]])),
-        (spec_histosys(), dict()),
+        (spec_histosys(), {}),
```
Comment (matthewfeickert) [inline tests/test_export.py:277]: This doesn't really do anything to justify the change, especially as since these are tests.

**Review by matthewfeickert — CHANGES_REQUESTED** [review]
Thanks for the PR. Though this PR doesn't really deal with code quality issues, but rather is just applying changes from the scan blindly. That's fine, but some of them don't really justify the noise they would add. 

If you take out the ones that I've flagged I think we'd accept this.

**Review by matthewfeickert — APPROVED** [review]
Thanks. LGTM :+1:

**Review by lukasheinrich — APPROVED** [review]

**Pull request merged**